near-abi-client/src/lib.rs

```diff
@@ -0,0 +1,83 @@
+use __private::{generate_abi_client, read_abi};
+use anyhow::{anyhow, Result};
+use convert_case::{Case, Casing};
+use quote::format_ident;
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+// Private functions shared between macro & generation APIs, not stable to be used.
+#[doc(hidden)]
+#[path = "private/mod.rs"]
+pub mod __private;
+
+/// Configuration options for ABI code generation.
+#[derive(Default)]
+pub struct Generator {
+    pub out_dir: Option<PathBuf>,
+    abis: Vec<(PathBuf, Option<String>)>,
+}
+
+impl Generator {
+    pub fn new(out_dir: PathBuf) -> Self {
+        Generator {
+            out_dir: Some(out_dir),
+            abis: vec![],
+        }
+    }
+
+    pub fn file(mut self, path: impl AsRef<Path>) -> Self {
```

Comment:
`Into<PathBuf>` feels like a better generic for this, so that the allocation can be re-used in some cases. (If someone passes in a `PathBuf` it currently takes a reference and then re-allocates